pkgs/tools/filesystems/dwarfs/default.nix

```diff
@@ -0,0 +1,99 @@
+{ stdenv
+, bison
+, boost
+, cmake
+, double-conversion
+, fetchFromGitHub
+, fmt_8
+, fuse3
+, gflags
+, git
+, glog
+, jemalloc
+, lib
+, libarchive
+, libdwarf
+, libevent
+, libunwind
+, lz4
+, openssl
+, pkgconfig
+, ronn
+, xxHash
+, xz
+, zstd
+}:
+
+stdenv.mkDerivation rec {
+  pname = "dwarfs";
+  version = "0.6.1";
+  src = fetchFromGitHub {
+    owner = "mhx";
+    repo = "dwarfs";
+    rev = "v${version}";
+    fetchSubmodules = true;
+    sha256 = "sha256-bGJkgcq8JxueRTX08QpJv1A0O5wXbiIgUY7BrY0Ln/M=";
+  };
+
+  # Needs tagged git repo for generating some files
+  # leaveDotGit alone isn't enough and breaks reproducible hashing
+  preConfigure = ''
+    git init
+    git add .
+    git -c user.name=root -c user.email=root@localhost commit -m "v${version}" -q
+    git -c user.name=root -c user.email=root@localhost tag "v${version}" -m "v${version}"
+  '';
+
+  # ...but the files are still kinda ugly, so let's make the data prettier for wherever it gets used ;)
+  postConfigure = ''
+    sed -i \
+      -e 's/\(PRJ_GIT_REV = \).*/\1"${src.rev}";/' \
+      -e 's/\(PRJ_GIT_DESC = \).*/\1"${version}";/' \
+      -e 's/\(PRJ_GIT_BRANCH = \).*/\1"${src.rev}";/' \
+      -e 's/\(PRJ_GIT_ID = \).*/\1"${version} from Nixpkgs :)";/' \
+      ../src/dwarfs/version.cpp
+  '';
+
+  cmakeFlags = [
+    "-DPREFER_SYSTEM_ZSTD=ON"
+    "-DPREFER_SYSTEM_XXHASH=ON"
+  ];
+
+  nativeBuildInputs = [
+    bison
+    boost.dev
```

Comment:
What is the rationale for including these dev outputs in `nativeBuildInputs`? I've never seen this before. Wouldn't you want the target system headers?

It's somewhat academic as Folly requires some extra work to cross compile.

Reply:
I'm somewhat of a newbie when it comes to Nix, so I wasn't entirely sure how to approach the dependencies. Having looked over the nixpkgs manual again I understand how it should be done now

Reply:
Loosely speaking `nativeBuildInputs` is for programs that must be executed at compile time, such as build tools and documentation generators. Everything needed at run time, including libraries and headers, goes in `buildInputs.`